core/container_authenticator.go

```diff
@@ -479,25 +481,41 @@ func (authenticator *ContainerAuthenticator) RequestToken() (*IamTokenServerResp
 // retrieveCRToken tries to read the CR token value from the local file system.
 func (authenticator *ContainerAuthenticator) retrieveCRToken() (crToken string, err error) {
 
-	// Use the default filename if one wasn't supplied by the user.
-	crTokenFilename := authenticator.CRTokenFilename
-	if crTokenFilename == "" {
-		crTokenFilename = defaultCRTokenFilename
+	if authenticator.CRTokenFilename != "" {
+		// Use the file specified by the user.
+		crToken, err = authenticator.readFile(authenticator.CRTokenFilename)
+	} else {
+		// If the user didn't specify a filename, try our two defaults.
+		crToken, err = authenticator.readFile(defaultCRTokenFilename1)
+		if err != nil {
```

Comment:
I think this is perfectly fine here, but a question came to my mind and can't let it go. We would like to try to read the 2nd file, even if the error is not a `FileNotFound` but - let's say - a permission issue, right? Really just asking... :)

Reply:
Good question... yes, I thought about the various errors that might occur and whether we should treat a permissions issue differently than a "file doesn't exist at all" issue.
I concluded that we should try to keep this simple and treat any error while trying to read the file the same... i.e. we couldn't read it (whatever the reason) so move on to the next one to try.  I think it's extremely unlikely that we'd encounter a scenario where both default files (vault-token and sa-token) exist and we end up using "sa-token" by mistake because "vault-token" is not readable by the application.